packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -230,7 +235,7 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     }
 
     final int lastIndex = indexOf(lastChild);
-    final double leadingScrollOffset = indexToLayoutOffset(itemExtent, firstIndex);
+    final double leadingScrollOffset = firstIndex == 0 ? 0 : indexToLayoutOffset(itemExtent, firstIndex - 1) + itemExtent;
```

Comment:
Not sure I understand the intentions here:

RenderSliverFillViewport.indexToLayoutOffset is:
```dart
return _padding + super.indexToLayoutOffset(itemExtent, index);
```
RenderSliverFixedExtendBoxAdaptor.indexToLayoutOffset (the superclass) is:
```dart
return itemExtent * index
```

So, ultimately, for end of this line becomes:
```
  indexToLayoutOffset(itemExtent, firstIndex - 1) + itemExtent
= _padding + super.indexToLayoutOffset(itemExtent, index - 1) + itemExtent
= _padding + (itemExtent * index - 1) + itemExtent
= _padding + itemExtend * index
= _padding + super.indexToLayoutOffset(itemExtent, index)
= indexToLayoutOffset(itemExtent, firstIndex)
```

So why do we need to subtract one from the index and then add another itemExtent?

Reply:
`indexToLayoutOffset(itemExtent, firstIndex - 1) + itemExtent` is more generic. It remains valid if there're gaps between items.